stdlib/public/Concurrency/Executor.swift

```diff
@@ -319,6 +461,192 @@ extension SerialExecutor {
+  }
+
+}
+
+
+/// The main executor must conform to these three protocols; we have to
+/// make this a protocol for compatibility with Embedded Swift.
+@available(SwiftStdlib 6.2, *)
+public protocol MainExecutor: RunLoopExecutor, SerialExecutor {
+}
+
+
+/// An ExecutorFactory is used to create the default main and task
+/// executors.
+@available(SwiftStdlib 6.2, *)
+public protocol ExecutorFactory {
+  #if !$Embedded
+  /// Constructs and returns the main executor, which is started implicitly
+  /// by the `async main` entry point and owns the "main" thread.
+  static var mainExecutor: any MainExecutor { get }
+  #endif
+
+  /// Constructs and returns the default or global executor, which is the
+  /// default place in which we run tasks.
+  static var defaultExecutor: any TaskExecutor { get }
+}
+
+@available(SwiftStdlib 6.2, *)
+@_silgen_name("swift_createExecutors")
+public func _createExecutors<F: ExecutorFactory>(factory: F.Type) {
+  #if !$Embedded && !SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
+  MainActor._executor = factory.mainExecutor
+  #endif
+  Task._defaultExecutor = factory.defaultExecutor
+}
+
+#if !$Embedded && !SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
+extension MainActor {
+  @available(SwiftStdlib 6.2, *)
+  static var _executor: (any MainExecutor)? = nil
+
+  /// The main executor, which is started implicitly by the `async main`
+  /// entry point and owns the "main" thread.
+  ///
+  /// Attempting to set this after the first `enqueue` on the main
+  /// executor is a fatal error.
+  @available(SwiftStdlib 6.2, *)
+  public static var executor: any MainExecutor {
+    if _executor == nil {
+      _executor = PlatformExecutorFactory.mainExecutor
+    }
+    return _executor!
+  }
+}
+#endif // !$Embedded && !SWIFT_STDLIB_TASK_TO_THREAD_MODEL_CONCURRENCY
+
+extension Task where Success == Never, Failure == Never {
+  @available(SwiftStdlib 6.2, *)
+  static var _defaultExecutor: (any TaskExecutor)? = nil
+
+  /// The default or global executor, which is the default place in which
+  /// we run tasks.
+  ///
+  /// Attempting to set this after the first `enqueue` on the global
+  /// executor is a fatal error.
+  @available(SwiftStdlib 6.2, *)
+  public static var defaultExecutor: any TaskExecutor {
+    if _defaultExecutor == nil {
+      _defaultExecutor = PlatformExecutorFactory.defaultExecutor
+    }
+    return _defaultExecutor!
+  }
+}
+
+extension Task where Success == Never, Failure == Never {
+  /// Get the current executor; this is the executor that the currently
+  /// executing task is executing on.
+  ///
+  /// This will return, in order of preference:
+  ///
+  ///   1. The custom executor associated with an `Actor` on which we are
+  ///      currently running, or
+  ///   2. The preferred executor for the currently executing `Task`, or
+  ///   3. The task executor for the current thread
+  ///
+  ///  If none of these exist, this property will be `nil`.
```

Comment:
This last line is no longer true; it will never be `nil`.

Reply:
I think that's reasonable hm... Let's see if SE review ends up with some more requests.

If not then we should probably clarify that a non-concurrency thread would end up returning the default executor - which seems fine